tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyThreadManager.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * -2010
+ */
+package org.unicode.cldr.web;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.concurrent.ManagedScheduledExecutorService;
+import javax.enterprise.concurrent.ManagedThreadFactory;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.apache.hc.core5.concurrent.DefaultThreadFactory;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRConfig.Environment;
+import org.unicode.cldr.util.CldrUtility;
+
+/**
+ * @author srl
+ *
+ *         A worker thread that performs various SurveyTool tasks, starting with
+ *         booting.
+ */
+public class SurveyThreadManager  {
+
+    private static final String JAVA_COMP_DEFAULT_MANAGED_THREAD_FACTORY = "java:comp/DefaultManagedThreadFactory";
+    private static final String DEFAULT_MANAGED_EXECUTOR = "java:comp/DefaultManagedExecutorService";
+    private static final String DEFAULT_MANAGED_SCHEDULED_EXECUTOR = "java:comp/DefaultManagedScheduledExecutorService";
+    static ThreadFactory gFactory = null;
+
+    static ExecutorService gExecutor = null;
+    static ScheduledExecutorService gScheduledExecutor = null;
+
+    public static ThreadFactory getThreadFactory() {
+        if(gFactory == null) {
```

Comment:
does getThreadFactory need to be public? it only seems to be called locally

also the callers are both lower in the file, so maybe this should be private and moved below the callers

Reply:
gExecutor could also be static?

Reply:
So, I'd like the ThreadFactory to be the preferred API (being standard) , and get rid of the homegrown addTask API. I'll do that.

Reply:
done